File: Backend/ExamSupportToolAPI/ExamSupportToolAPI.DataObjects/StudentPresentationForCommittee.cs
namespace ExamSupportToolAPI.DataObjects
{
    public class StudentPresentationForCommittee
    {
        public Guid Id { get; set; }
        public Boolean IsAbsent { get; set; }
        public DateTime StartingTime { get; set; }
        public StudentPresentationStatus Status { get; set; }
        public StudentForCommittee? Student { get; set; }
        public ExaminationTicketForCommittee? ExaminationTicket { get; set; }
        public int StudentPresentationDuration { get; set; }

        public Guid CurrentCommitteeId { get; set; }

    }
    public enum StudentPresentationStatus
    {
        Created,
        Presenting,
        Completed,
        Graded
    }
}

File: Frontend/ExamSupportToolUI/src/app/shared/models/committee/student-presentation-for-committee.ts
import { StudentPresentationStatus } from "../student-presentation-status";
import { ExaminationTicketForCommittee } from "./examination-ticket-for-committee";
import { StudentForCommittee } from "./student-for-committee";


export interface StudentPresentationForCommittee {
    id: string;
    isAbsent: boolean;
    startingTime?: Date;
    status: StudentPresentationStatus;
    student: StudentForCommittee | null;
    examinationTicket: ExaminationTicketForCommittee | null;
    studentPresentationDuration: number;
    currentCommitteeId: string;
}
